beacon_chain/types/src/validator_record.rs

```diff
@@ -35,61 +31,69 @@ pub struct ValidatorRecord {
     pub withdrawal_credentials: Hash256,
     pub randao_commitment: Hash256,
     pub randao_layers: u64,
-    pub status: ValidatorStatus,
-    pub latest_status_change_slot: u64,
+    pub activation_slot: u64,
+    pub exit_slot: u64,
+    pub withdrawal_slot: u64,
+    pub penalized_slot: u64,
     pub exit_count: u64,
+    pub status_flags: Option<StatusFlags>,
```

Comment:
We could internally call anything not 1 or 2 some `Undefined` variant and then everything is sound

Reply:
But I agree if we can change how it is handled in the spec then it will be clearer to implement.

Do you like having this field be "nullable"? It can either stay typed `Option<Flag>` or go to `Flag` and the spec defines a particular value. If we keep it nullable, then we should use `None` rather than `0` in those places it is used that way.

Reply:
I think the `Option` is good whilst it's undefined.

We could risk a consensus split if we accept arbitrary values when other clients don't. IMO the EF should clearly define `0` and specify that no other value is valid.